lib/erlef/token.ex

```diff
@@ -0,0 +1,36 @@
+defmodule Erlef.Token do
+  @moduledoc """
+  Erlef.Token serializer and deserializer
+  """
+  @aad "AES256GCM"
```

Comment:
fwiw AES-GCM-256 recommends a rotation of keys at "frequent" intervals. From NIST:

>  The following requirement applies to all implementations that use either 1) the deterministic construction with IVs whose length is not 96, or 2) the RBG-based construction, for IVs of any length. In other words, unless an implementation only uses 96-bit IVs that are generated by the deterministic construction:
> 
> The total number of invocations of the authenticated encryption function shall not exceed 2^32, including all IV lengths and all instances of the authenticated encryption function with the given key.
> 
> This is a “global” requirement that can be achieved by appropriate “local” limits on each instance of the authenticated encryption function with a given key. For example, suppose an implementation consists of 2^10 devices that only support 64-bit, 96-bit, and 128-bit IVs. One way to satisfy the above requirement would be to limit each device to 2^20 invocations with 64-bit IVs, 2^21 invocations with 96-bit IVs, and 2^20 invocations with 128-bit IVs.

This implementation uses Erlang's crypto:strong_rand_bytes(16) to generate 128-bit IVs.

As such, it would be recommended to rotate the key used to encrypt content once such limits (2^21 calls locally, or 2^32 overall) are reached to avoid weakening the security of the private key.

Reply:
So, Erlef.Token is not in play yet, it may be will get ripped out of this PR. But my initial thinking was we can do better than AES-GCM-256, but perhaps might be overkill, so was going to save that idea for this review... and here we are 😁  What do you think about a stronger algo and perhaps one that does not exist in crypto? Namely, https://libsodium.gitbook.io/doc/secret-key_cryptography/aead/chacha20-poly1305/xchacha20-poly1305_construction

Reply:
Let me add that rotating the keys is important, as the current implementation of Erlef.Token is pretty much if not exactly what Plug.Crypto does for us now.

Reply:
I would also invite thoughts from @voltone and others from the security working group on all this as well. Simply can not get enough 👀 on these matters.

Reply:
AES-GCM-256 is plenty strong for a lot of uses. You can probably do some napkin math based on average # of visits vs. max encryptions to pick when to just change the key and invalidate a bunch of things. It's not critical at this point but for anything higher volume it can become so.

Reply:
aye 👍

Reply:
Yeah, just dropped it for now. But yes, it implement a Plug.Session. We'll circle back to it for sure though.